lineapy/graph_reader/artifact_collection.py

```diff
@@ -61,8 +62,19 @@ def __init__(
 
         artifacts_by_session: Dict[LineaID, List[LineaArtifact]] = {}
 
+        artifact_names = [
+            art if isinstance(art, str) else art[0] for art in artifacts
+        ]
+        artifacts_and_caches = artifacts + [
```

Comment:
think about renaming caches here, im thinking maybe we should say this is `input_and_output_artifacts` or just call it `all_artifacts` and rename `artifacts` to `output_ artifacts` and `caches` to `input_artifacts`